docs/docs/external-volumes

```diff
@@ -0,0 +1,81 @@
+
+Using an external storage service allows your apps to be more fault-tolerant. If a host fails, Marathon reschedules it on another host, along with its associated data, without user intervention.
+
+# Specifying an External Volume
+
+When you install DCOS, add the following to your `genconf.yml` file:
+
+- `rexray_config_method: file`
+
+- `rexray_config_filename: /path/to/rexray.yaml`
+
+<!-- [Learn more](URL). link to installer docs that mention this when they are live-->
+
+## Scaling your App
+
+Apps that use external volumes should only be scaled to a single instance.
+
+If you scale your app down to 0 instances, the volume is unattached from the agent where it was mounted, but it is not deleted. If you scale your app up again, the data that had been associated with it will still be available.
+
+## Create an Application with External Volumes
+
+You specify an external volume in the app definition of your Marathon app. [Learn more about Marathon application definitions](application-basics.html).
+
+```json
+{ 
+  "id": "hello",
+     "cmd": "while [ true ] ; do touch/var/lib/rexray/volumes/test12345/data/hello ; sleep 5 ; done",   
+     "mem": 32,   
+     "cpus": 0.1,   
+     "instances": 1,
+     "env": {
+          "DVDI_VOLUME_DRIVER": "rexray",
+          "DVDI_VOLUME_NAME": "test12345",
+          "DVDI_VOLUME_OPTS":"size=1,volumetype=standard,newfstype=ext4,overwritefs=false"
+     }
+}
+```
+
+In the app definition above, the environment variables (`env`) specify a 1GB volume that will be formatted ext4 once it is created.
+
+- Volumes will always be available under `/var/lib/rexray/volumes/$DVDI_VOLUME_NAME/`. **Note:** This applies only to non-Docker apps; Docker apps can specify the mount point inside the app definition.
+
+- The root path inside the volume for storing data is configurable, but defaults to `data/`. See [the REX-Ray documentation](https://rexray.readthedocs.org/en/v0.3.2/user-guide/config/#volume-root-path-031) for more information.
+
+- `DVDI_VOLUME_DRIVER` tells `DVDCLI` which Docker volume driver we want to use for storage. The only driver we currently provide is REX-Ray, so this value should always be `rexray`. [Learn more about REX-ray](https://rexray.readthedocs.org/en/v0.3.2/user-guide/schedulers/).
+
+- `DVDI_VOLUME_NAME` is the name by which REX-Ray looks up your volume. When your task is staged on an agent, REX-Ray queries the storage service for a volume with this name. If one does not exist, it's created. Otherwise, the existing volume is re-used.
+
+- `DVDI_VOLUME_OPTS` is the set of volume options passed to DVDCLI. [See the full list of options](https://github.com/emccode/dvdcli/blob/3c3dcbe1490c2c4a150555978f2b4915dafaddf2/README.md#extra-options).
+
+You can define multiple volumes per app by repeating the environment variables with 0, 1 ,2, etc. appended to them. For example, `DVDI_VOLUME_NAME0`.
```

Comment:
Thanks for clarifying!

On Wed, Mar 30, 2016 at 7:51 PM, Branden Rolston <notifications@github.com>
wrote:

> In docs/docs/external-volumes
> <https://github.com/mesosphere/marathon/pull/3607#discussion_r57982761>:
>
> > +}
> > +```
> > +
> > +In the app definition above, the environment variables (`env`) specify a 1GB volume that will be formatted ext4 once it is created.
> > +
> > +- Volumes will always be available under `/var/lib/rexray/volumes/$DVDI_VOLUME_NAME/`. **Note:** This applies only to non-Docker apps; Docker apps can specify the mount point inside the app definition.
> > +
> > +- The root path inside the volume for storing data is configurable, but defaults to `data/`. See [the REX-Ray documentation](https://rexray.readthedocs.org/en/v0.3.2/user-guide/config/#volume-root-path-031) for more information.
> > +
> > +- `DVDI_VOLUME_DRIVER` tells `DVDCLI` which Docker volume driver we want to use for storage. The only driver we currently provide is REX-Ray, so this value should always be `rexray`. [Learn more about REX-ray](https://rexray.readthedocs.org/en/v0.3.2/user-guide/schedulers/).
> > +
> > +- `DVDI_VOLUME_NAME` is the name by which REX-Ray looks up your volume. When your task is staged on an agent, REX-Ray queries the storage service for a volume with this name. If one does not exist, it's created. Otherwise, the existing volume is re-used.
> > +
> > +- `DVDI_VOLUME_OPTS` is the set of volume options passed to DVDCLI. [See the full list of options](https://github.com/emccode/dvdcli/blob/3c3dcbe1490c2c4a150555978f2b4915dafaddf2/README.md#extra-options).
> > +
> > +You can define multiple volumes per app by repeating the environment variables with 0, 1 ,2, etc. appended to them. For example, `DVDI_VOLUME_NAME0`.
>
> Any single digit is an acceptable suffix. No suffix is equivalent to a
> suffix of 0.
> https://github.com/emccode/mesos-module-dvdi/blob/9f4ab1cf187fb7b8ba6b0ec5b5518860cdb8ee44/isolator/isolator/docker_volume_driver_isolator.cpp#L491
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/mesosphere/marathon/pull/3607/files/509bd2c457d883211a03ff4d09098a59914ac5cb#r57982761>
>